zebrad/tests/acceptance.rs

```diff
@@ -461,3 +461,80 @@ fn valid_generated_config(command: &str, expected_output: &str) -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn metrics_tracing_listeners() -> Result<()> {
+    use hyper::{Body, Client, Request, Uri};
+
+    zebra_test::init();
+
+    // From here we dont know what port the OS selected for us if we use port 0 so we use a hard coded port for the 2 endpoints.
+    let metrics_endpoint = "127.0.0.1:9998";
+    let metrics_url = "http://127.0.0.1:9998";
+    let tracing_endpoint = "127.0.0.1:9999";
+    let tracing_url = "http://127.0.0.1:9999/filter";
+
+    // Write a configuration that has both metrics endpoint_addr and tracing endpoint_addr options set
+    let mut config = default_test_config()?;
+    config.metrics.endpoint_addr = Some(metrics_endpoint.parse().unwrap());
+    config.tracing.endpoint_addr = Some(tracing_endpoint.parse().unwrap());
+
+    let dir = TempDir::new("zebrad_tests")?;
+    fs::File::create(dir.path().join("zebrad.toml"))?
+        .write_all(toml::to_string(&config)?.as_bytes())?;
+
+    let tempdir = dir.path().to_path_buf();
+
+    let mut child = get_child(&["start"], &tempdir)?;
+
+    // Run the program for a second before testing the endpoints
+    std::thread::sleep(Duration::from_secs(1));
+
+    // Create an http client
+    let client = Client::new();
+
+    // Test metrics endpoint
+    let metrics_res = client.get(Uri::from_static(metrics_url)).await?;
+    assert!(metrics_res.status().is_success());
+    let metrics_body = hyper::body::to_bytes(metrics_res).await?;
+    assert!(std::str::from_utf8(&metrics_body)
+        .unwrap()
+        .contains("metrics snapshot"));
+
+    // Test tracing endpoint
+    let tracing_res = client.get(Uri::from_static(tracing_url)).await?;
+    assert!(tracing_res.status().is_success());
+    let tracing_body = hyper::body::to_bytes(tracing_res).await?;
+    assert!(std::str::from_utf8(&tracing_body).unwrap().contains("info"));
+    // Set a filter and make sure it was changed
+    let request = Request::post(tracing_url)
+        .body(Body::from("zebrad=trace"))
```

Comment:
Trace might produce lots of output, let's just use debug?
```suggestion
        .body(Body::from("zebrad=debug"))
```

Reply:
Done as part of https://github.com/ZcashFoundation/zebra/pull/1000/commits/e301ab938643fe8369752e9228fda5094699efbb